### DISCUSSION TITLE
Why is the SDK try to connect to both IPv4 and IPv6 addresses of iotcore endpoint simultaneously ?

### DISCUSSION THREAD

**Discussion by kiranpradeep** [aws/aws-iot-device-sdk-js-v2#481] [Q&A] [answered]
I am using aws-iot-device-sdk-v2 with Node.js. I was inspecting logs obtained after enabling // io.enable_logging(io.LogLevel.DEBUG); //. I see that SDK after resolving host to IPs, is trying to connect to both IPv4 and IPv6 addresses **simultaneously**. This is surprising. Most software does not do this.

1. Is this behavior that can be controlled? E.g. If I prefer the SDK to connect to a single IP and return with failure if connect fails.
2. I also like to understand, what is the scenario that benefits from this kind of behavior.

I have attached [logs.txt](https://github.com/aws/aws-iot-device-sdk-js-v2/files/14636709/logs.txt)

**Comment by bretambrose** [comment, marked as answer]
This is intentional.  The short answer is a strong push for happy eyeballs support: https://datatracker.ietf.org/doc/html/rfc6555

There is no way to disable this behavior.
